src/components/search_bar/search_bar.tsx

```diff
@@ -40,25 +43,33 @@ class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
         }
     }
 
+    componentDidUpdate(prevProps: SearchBarProps, prevState: SearchBarState) {
+        const { query } = this.props;
+        if (query !== prevProps.query && query !== prevState.query) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({
+                query,
+            });
+        }
+    }
+
     componentWillUnmount() {
```

Comment:
Would it be fine to replace it with `componentDidUpdate()`?
https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate

Reply:
This is `componentWillUnmount`, not `componentWillUpdate`